msg/SemanticSegmentation.msg

```diff
@@ -0,0 +1,21 @@
+# A message to contain the output of a Semantic Segmentation process
+
+std_msgs/Header header
+
+uint32 height         # mask image height, that is, number of rows
+uint32 width          # mask image width, that is, number of columns
+
+# the bytes of the single-channel image made by
+# the segmentation mask
+uint8[] data # 
+
+# the confidence of the inference of each pixel
+# between 0-100%
+uint8[] confidence 
```

Comment:
That's not standard for the output of AI libraries, they usually give out a probability value. It would be a bit unnatural to convert it to 0-255 for use (e.g. I want to use predictions that are at least 80% confident)

Reply:
It could be a float, but does it really make a difference to have a confidence of say 60% (uint8) or 60.5% (float) knowing that this would increase the space used by this array by 4?

Reply:
Floats are the same size

Reply:
Specially on an array field that is expected to contain a lot of elements. On a 640x360 mask changing this for a float32 would mean an increase of 640x360x3 bytes, almost 700kB per message with respect to uint8

Reply:
You should make sure to explain these are uints not floats, otherwise people will just put their `0-1` percentages from TF or something into it and it'll always render as `0` since it needs to be multiplied by 100 and then cast to an int